/*
 * This file was automatically generated by EvoSuite
 * Mon May 25 08:25:29 GMT 2020
 */

package ElevatorSystem;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import ElevatorSystem.Elevator;
import ElevatorSystem.Environment;
import ElevatorSystem.EvilPerson;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class EvilPerson_ESTest extends EvilPerson_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      int[] intArray0 = new int[1];
      intArray0[0] = 1339;
      Environment environment0 = new Environment(1339);
      EvilPerson evilPerson0 = new EvilPerson("", 1339, 0, 0, intArray0, environment0);
      Elevator elevator0 = new Elevator(environment0, false);
      // Undeclared exception!
      try { 
        evilPerson0.enterElevator(elevator0);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // 1339
         //
         verifyException("ElevatorSystem.Elevator", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      int[] intArray0 = new int[4];
      EvilPerson evilPerson0 = null;
      try {
        evilPerson0 = new EvilPerson("Pl|GuiF9", 1, 1, 1, intArray0, (Environment) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("ElevatorSystem.Person", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      int[] intArray0 = new int[8];
      Environment environment0 = new Environment(0);
      EvilPerson evilPerson0 = null;
      try {
        evilPerson0 = new EvilPerson("6>ign", (-597), (-2881), 440, intArray0, environment0);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // -2881
         //
         verifyException("ElevatorSystem.Environment", e);
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      int[] intArray0 = new int[1];
      Environment environment0 = new Environment(127);
      EvilPerson evilPerson0 = new EvilPerson("K#$", 127, 21, 21, intArray0, environment0);
      Elevator elevator0 = new Elevator(environment0, false);
      evilPerson0.enterElevator(elevator0);
      assertEquals(0, elevator0.getCurrentFloorID());
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      Environment environment0 = new Environment(891);
      EvilPerson evilPerson0 = new EvilPerson("", 648, 648, 0, (int[]) null, environment0);
      Elevator elevator0 = new Elevator(environment0, true);
      // Undeclared exception!
      try { 
        evilPerson0.enterElevator(elevator0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("ElevatorSystem.EvilPerson", e);
      }
  }
}
